prospero-common/src/main/java/org/wildfly/prospero/actions/MetadataActions.java

```diff
@@ -0,0 +1,64 @@
+package org.wildfly.prospero.actions;
+
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+
+import org.wildfly.prospero.Messages;
+import org.wildfly.prospero.api.InstallationMetadata;
+import org.wildfly.prospero.api.exceptions.MetadataException;
+import org.wildfly.prospero.model.ProvisioningConfig;
+import org.wildfly.prospero.model.RepositoryRef;
+
+/**
+ * Metadata related actions wrapper.
+ */
+public class MetadataActions {
+
+    private final Path installation;
+
+    public MetadataActions(Path installation) {
+        this.installation = installation;
+    }
+
+    /**
+     * Adds a remote maven repository to an installation.
+     */
+    public void addRepository(String name, URL url) throws MetadataException {
+        InstallationMetadata installationMetadata = new InstallationMetadata(installation);
+        ProvisioningConfig prosperoConfig = installationMetadata.getProsperoConfig();
+        if (prosperoConfig.addRepository(new RepositoryRef(name, url.toString()))) {
+            installationMetadata.updateProsperoConfig(prosperoConfig);
+        } else {
+            throw Messages.MESSAGES.repositoryExists(name, url);
+        }
+    }
+
+    /**
+     * Removes a remote maven repository to an installation.
+     */
+    public void removeRepository(String id) throws MetadataException {
+        InstallationMetadata installationMetadata = new InstallationMetadata(installation);
+        ProvisioningConfig prosperoConfig = installationMetadata.getProsperoConfig();
+        Optional<RepositoryRef> repo = prosperoConfig.getRepositories().stream()
```

Comment:
Can you put this as a `removeRepository` method in `ProvisioninConfig`? I think this will be used in other places as well.

btw. I think `ProvisioninConfig` should be renamed to `ProsperoConfig`, but that's separate PR, WDYT?

Reply:
+1 for renaming, wanted to suggest it too. Also thought about renaming the `InstallationMetadata#public org.jboss.galleon.config.ProvisioningConfig getProvisioningConfig()` to `#getGalleonProvisioningConfig()`, to distinguish it more clearly in the API.

Reply:
I'll make those changes when we merge open PRs to avoid conflicts